books/calculus/bake

```diff
@@ -0,0 +1,278 @@
+    heading = chapter.at('h1[2]')
+    heading[:id] = "chapTitle#{chapter.count_in(:book)}"
+    heading.replace_children(with:
+      <<~HTML
+        <span class="os-part-text">Chapter </span>
+        <span class="os-number">#{chapter.count_in(:book)}</span>
+        <span class="os-divider"> </span>
+        <span data-type="" itemprop="" class="os-text">#{heading.text}</span>
+      HTML
+    )
+
+    chapter.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table,
+                           number: "#{chapter.count_in(:book)}.#{table.count_in(:chapter)}")
+    end
+
+    chapter.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: "#{chapter.count_in(:book)}.#{example.count_in(:chapter)}",
+                     title_tag: 'h3')
+
+      example.search('div[data-type="exercise"]').each do |exercise|
+        BakeExercises.example_exercises(exercise: exercise)
+      end
+    end
+
+    chapter.pages('$:not(.introduction)').each do |page|
+      page.search("div[data-type='description']").each(&:trash)
+      page.add_class('chapter-content-module')
+
+      title = page.search("div[data-type='document-title']").first
+      title.name = 'h2'
+      title.replace_children(with:
+        <<~HTML
+          <span class="os-number">#{chapter.count_in(:book)}.#{page.count_in(:chapter)}</span>
+          <span class="os-divider"> </span>
+          <span data-type="" itemprop="" class="os-text">#{title.text}</span>
+        HTML
+      )
+    end
+
+    chapter.figures.each do |figure|
+      BakeFigure.v1(figure: figure,
+                    number: "#{chapter.count_in(:book)}.#{figure.count_in(:chapter)}")
+    end
+  end
+
+  book.pages('$.appendix').each do |page|
+    appendix_letter = [*('A'..'Z')][page.count_in(:book) - 1]
+
+    page.figures.each do |figure|
+      BakeFigure.v1(figure: figure, number: "#{appendix_letter}#{figure.count_in(:page)}")
+    end
+
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table, number: "#{appendix_letter}#{table.count_in(:page)}")
+    end
+
+    page.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: "#{appendix_letter}#{example.count_in(:page)}",
+                     title_tag: 'div')
+    end
+
+    BakeAppendix.v1(page: page, number: appendix_letter)
+  end
+
+  BakeStepwise.v1(book: book)
+  BakeUnnumberedTables.v1(book: book)
+
+  book.pages.each do |page|
+    # fix table captions
+
+    page.search('.section-exercises').each do |section|
+      section.search('.os-problem-container > .os-table:first-child').each do |table|
+        table.add_class('first-element')
+        table.parent.add_class('has-first-element')
+      end
+
+      section.search('.os-problem-container > [data-type="media"]:first-child').each do |img|
+        img.add_class('first-element')
+        img.parent.add_class('has-first-element')
+      end
+    end
```

Comment:
Add a `BakeProblemFirstElements`?  Does it even need the `page.search(...) do |section|` -- can it instead just have `book.search('.os-problem-container > :first-child').each do |element| element.add_class(...) ...`

Reply:
Can we do this?  Even if we don't have `BakeProblemFirstElements` we could maybe do

```ruby
book.search('.os-problem-container > :first-child').each do |element|
  element.add_class('first-element')
  element.parent.add_class('has-first-element')
end
```